pkg/beholder/client.go

```diff
@@ -100,6 +104,11 @@ func NewGRPCClient(cfg Config, otlploggrpcNew otlploggrpcFactory) (*Client, erro
 		otlploggrpc.WithEndpoint(cfg.OtelExporterGRPCEndpoint),
 	}
 	// Initialize auth here for reuse with log, trace, and metric exporters
+	// Two modes are supported:
+	// 1. Rotating auth: If AuthKeySigner is set, AuthHeaders are used as initial headers
+	//    that will be rotated by the lazy signer after TTL expires
+	// 2. Static auth: If AuthKeySigner is nil, AuthHeaders are used as-is and never change
+	lazySigner := NewLazySigner()
```

Comment:
It looks like the zero value is usable, so the constructor could be dropped:
```suggestion
	var signer lazySigner
```